pubtools/pulplib/_impl/model/repository/base.py

```diff
@@ -595,6 +597,50 @@ def sync(self, options=None):
             )
         )
 
+    def lock(self, context, duration=None):
+        """
+        Obtain an exclusive advisory lock on this repository.
+
+        Returns a context manager representing the lock, intended to be used
+        via a `with` statement. When the context is entered, the caller will
+        wait until the lock can be acquired (or raise an exception if the lock
+        can't be acquired).
+
+        Only a single :class:`~pubtools.pulplib.Client` is able to hold the lock
+        on a repository at any given time. The lock does not prevent modifications
+        to the repo with the Pulp API, and does not affect other Pulp client
+        implementations or instances of :class:`~pubtools.pulplib.Client` not
+        using the `lock` method.
+        Args:
```

Comment:
Needs a blank line before it:

```suggestion

        Args:
```